src/specific/s_audio_stream.c

```diff
@@ -516,3 +521,41 @@ void S_Audio_StreamSoundMix(float *dst_buffer, size_t len)
         }
     }
 }
+
+int64_t S_Audio_StreamGetTimestamp(int sound_id)
+{
+    if (!g_AudioDeviceID || sound_id < 0
+        || sound_id >= AUDIO_MAX_ACTIVE_STREAMS) {
+        return -1;
+    }
+
+    if (m_StreamSounds[sound_id].is_playing) {
+        SDL_LockAudioDevice(g_AudioDeviceID);
+        AUDIO_STREAM_SOUND *stream = &m_StreamSounds[sound_id];
+        int64_t timestamp = stream->timestamp;
+        SDL_UnlockAudioDevice(g_AudioDeviceID);
+        return timestamp;
+    }
+
+    return -1;
+}
+
+bool S_Audio_StreamSeekTimestamp(int sound_id, int64_t timestamp)
+{
+    if (!g_AudioDeviceID || sound_id < 0
+        || sound_id >= AUDIO_MAX_ACTIVE_STREAMS) {
+        return false;
+    }
+
+    if (m_StreamSounds[sound_id].is_playing) {
+        SDL_LockAudioDevice(g_AudioDeviceID);
+        AUDIO_STREAM_SOUND *stream = &m_StreamSounds[sound_id];
+        av_seek_frame(
+            stream->av.format_ctx, sound_id, timestamp, AVSEEK_FLAG_ANY);
+        avcodec_flush_buffers(stream->av.codec_ctx); // ?
```

Comment:
? :D

Reply:
Lmao I tried to fix the audio clipping for a frame or two on load, but this didn't help. From what I read though, the buffer should be flushed after seeking. There doesn't seem to be the most online resources for this though, so it's hard to find help and ideas.

http://dranger.com/ffmpeg/tutorial07.html

Reply:
I don't see anything about flushing the buffers in ffmpeg documentation. (They have excellent documentation in their source code.) So if this doesn't help us any way let's remove this call.